src/wasm.cc

```diff
@@ -447,6 +447,32 @@ void WasmBase::finishShutdown() {
   }
 }
 
+bool WasmHandleBase::canary(const std::shared_ptr<PluginBase> &plugin,
+            const WasmHandleCloneFactory &clone_factory) {
+  auto configuration_canary_handle = clone_factory(shared_from_this());
+  if (!configuration_canary_handle) {
+    this->wasm()->fail(FailState::UnableToCloneVm, "Failed to clone Base Wasm");
```

Comment:
AIUI this is calling fail() on a WasmBase instance that may already be in use by an earlier caller of createWasm()--does that raise any issues? @PiotrSikora 

Maybe in practice this is very unlikely, since that earlier caller of createWasm() would have gone through this codepath as well, so if clone_factory were broken, it would have failed then as well.

Reply:
I am not 100% sure, but AFAIK, the earlier caller of the createWasm() should rely on the thread-specific Wasm VM. Therefore, I think that the earlier caller is not affected by calling "fail()" later. 
@PiotrSikora WDYT?

(BTW, as we know, this is the code piece which was already here.)